src/app/CommandHandler.h

```diff
@@ -50,8 +53,33 @@ class CommandHandler : public Command
 
 private:
     friend class TestCommandInteraction;
+    friend class CommandHandlerHandleDeleter;
     CHIP_ERROR SendCommandResponse();
     CHIP_ERROR ProcessCommandDataElement(CommandDataElement::Parser & aCommandElement) override;
+    static void FinishPendingWork(System::Layer * aLayer, void * aAppState, CHIP_ERROR aError);
+
+    struct CommandHandlerHandleDeleter
+    {
+        void operator()(CommandHandler * apCommandHandler) const;
+    };
+
+    uint32_t mPendingWorkCount = 0;
+
+public:
+    using CommandHandlerAsyncHandle = std::unique_ptr<CommandHandler, CommandHandlerHandleDeleter>;
```

Comment:
I'm uneasy with the alias hiding away the fact that this is a unique_ptr. We should not try to hide the ownership semantics here of this handle.

The other detail that is un-obvious is the fact that assigning 'nullptr' to that handle 'releases' that, and causes a response to get sent out.

Reply:
Sending out messages on destructors just does not feel right.